build.gradle

```diff
@@ -153,158 +157,205 @@ publishing {
+         * The private GPG (ascii armored) used to sign files and commits.
+         * Required when mode = `MEMORY` || `FILE`.
+         * Set via env var `JRELEASER_GPG_SECRET_KEY`.
+         *
+         * NOTE: The name of the project property we set this to is derived from our GitHub
+         * publishing workflow, which sets the project property using an environment variable.
+         * NOTE: If set to a non-empty malformed value, signing will fail with "Unexpected error".
+         */
         secretKey = findProperty("signingKey")
     }
 
-    // Artifact deployment configuration.
+    /** Artifact deployment configuration. */
     deploy {
-        // Deploy staged artifacts to target Maven compatible repositories.
+        /** Deploy staged artifacts to target Maven compatible repositories. */
         maven {
-            // Deploy staged artifacts to Maven Central using the publisher API.
-            // See https://jreleaser.org/guide/latest/reference/deploy/maven/maven-central.html
+            /**
+             * Deploy staged artifacts to Maven Central using the publisher API.
+             * See https://jreleaser.org/guide/latest/reference/deploy/maven/maven-central.html
+             */
             mavenCentral {
-                // The name of the deployer (deployers require a name).
+                /** The name of the deployer (deployers require a name). */
                 app {
-                    // Enables or disables the deployer.
-                    // Supported values are [`NEVER`, `ALWAYS`, `RELEASE`, `SNAPSHOT`].
-                    // Defaults to `NEVER`.
+                    /**
+                     * Enables or disables the deployer.
+                     * Supported values are [`NEVER`, `ALWAYS`, `RELEASE`, `SNAPSHOT`].
+                     * Defaults to `NEVER`.
+                     */
                     active = 'ALWAYS'
 
-                    // URL where the MavenCentral service is enabled.
+                    /** URL where the MavenCentral service is enabled. */
                     url = 'https://central.sonatype.com/api/v1/publisher'
 
-                    // The username required for authorization.
-                    // Set via env var `JRELEASER_MAVENCENTRAL_USERNAME`.
-                    //
-                    // NOTE: We set this using a project property. See
-                    // https://docs.gradle.org/current/userguide/build_environment.html#sec:project_properties
+                    /**
+                     * The username required for authorization.
+                     * Set via env var `JRELEASER_MAVENCENTRAL_USERNAME`.
+                     *
+                     * NOTE: The name of the project property we set this to is derived from our
+                     * GitHub publishing workflow, which sets the project property using an
+                     * environment variable.
+                     */
                     username = findProperty("sonatypeUsername")
```

Comment:
As per comment on ticket, let's get rid of Sonatype references (including updating our `github-actions` repo.
Go with:

```
username = findProperty("mavenCentralUsername")
password = findProperty("mavenCentralPassword")